src/lib/CoreInstance.cc

```diff
@@ -4,70 +4,52 @@ namespace simeng {
 
 CoreInstance::CoreInstance(std::shared_ptr<memory::MMU> mmu,
                            arch::sendSyscallToHandler handleSyscall)
-    : config_(Config::get()), mmu_(mmu), handleSyscall_(handleSyscall) {
-  setSimulationMode();
+    : config_(config::SimInfo::getConfig()),
+      mmu_(mmu),
+      handleSyscall_(handleSyscall) {
   createCore();
 }
 
-void CoreInstance::setSimulationMode() {
-  // Get the simualtion mode as defined by the set configuration, defaulting to
-  // emulation
-  if (config_["Core"]["Simulation-Mode"].as<std::string>() ==
-      "inorderpipelined") {
-    mode_ = SimulationMode::InOrderPipelined;
-    modeString_ = "In-Order Pipelined";
-  } else if (config_["Core"]["Simulation-Mode"].as<std::string>() ==
-             "outoforder") {
-    mode_ = SimulationMode::OutOfOrder;
-    modeString_ = "Out-of-Order";
-  }
-  return;
-}
-
 void CoreInstance::createCore() {
   // Create the architecture, with knowledge of the OS
-  if (config_["Core"]["ISA"].as<std::string>() == "rv64") {
+  if (config::SimInfo::getISA() == config::ISA::RV64) {
     arch_ = std::make_unique<simeng::arch::riscv::Architecture>();
-  } else if (config_["Core"]["ISA"].as<std::string>() == "AArch64") {
+  } else if (config::SimInfo::getISA() == config::ISA::AArch64) {
     arch_ = std::make_unique<simeng::arch::aarch64::Architecture>();
   }
 
   // Construct branch predictor object
   predictor_ = std::make_unique<simeng::GenericPredictor>();
 
-  // Extract port arrangement from config file
+  // Extract the port arrangement from the config file
   auto config_ports = config_["Ports"];
-  std::vector<std::vector<uint16_t>> portArrangement(config_ports.size());
-  for (size_t i = 0; i < config_ports.size(); i++) {
-    auto config_groups = config_ports[i]["Instruction-Group-Support"];
+  std::vector<std::vector<uint16_t>> portArrangement(
+      config_ports.num_children());
+  for (size_t i = 0; i < config_ports.num_children(); i++) {
+    auto config_groups = config_ports[i]["Instruction-Group-Support-Nums"];
     // Read groups in associated port
-    for (size_t j = 0; j < config_groups.size(); j++) {
-      portArrangement[i].push_back(config_groups[j].as<uint16_t>());
+    for (size_t j = 0; j < config_groups.num_children(); j++) {
+      portArrangement[i].push_back(
+          config::SimInfo::getValue<uint16_t>(config_groups[j]));
     }
   }
   portAllocator_ = std::make_unique<simeng::pipeline::BalancedPortAllocator>(
       portArrangement);
```

Comment:
You are correct, is needed by the new inorder core now. Given the upcoming planned changes to remove CoreInstance I don't think the conditional setting of the portAllocator is necessary. I can change it if you feel strongly about it but will all be changed in due course. With the new core factory approach, conditions akin to this would be present.

Reply:
Not a major issue and especially if all is changing can leave as is